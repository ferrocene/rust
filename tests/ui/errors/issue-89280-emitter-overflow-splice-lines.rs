//@ revisions: edition2015 edition2018
//@ [edition2015] edition: 2015
//@ [edition2018] edition: 2018..
//@ [edition2015] check-pass

trait X {
    fn test(x: u32, (
//[edition2015]~^ WARN anonymous parameters are deprecated and will be removed in the next edition
//[edition2015]~^^ WARN this is accepted in the current edition (Rust YYYY) but is a hard error in Rust 2018!
    )) {}
//[edition2018]~^ ERROR expected one of `:` or `|`, found `)`
}

fn main() {}
