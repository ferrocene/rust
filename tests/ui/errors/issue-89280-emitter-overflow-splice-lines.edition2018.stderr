error: expected one of `:` or `|`, found `)`
  --> $DIR/issue-89280-emitter-overflow-splice-lines.rs:10:6
   |
LL |     )) {}
   |      ^ expected one of `:` or `|`
   |
   = note: anonymous parameters are removed in the 2018 edition (see RFC 1685)
help: explicitly ignore the parameter name
   |
LL |     fn test(x: u32, _: (
   |                     ++

error: aborting due to 1 previous error

